docs/DeveloperGuide.md

```diff
@@ -427,15 +446,106 @@ object.
+all the `leftOverModules` and a `RequiredCommandStorage` to store all modules in the syllabus.
 
 7. Then, `RequiredCommand.execute()` retrieves the `GradPad` object stored within `Model` and accesses the `modules
-` field within the `GradPad`.
+` field within the `GradPad` with a few method calls, before storing it in `currentModules`.
+
+8. `RequiredCommand.execute()` then calls its own method `setStorage` to create a `RequiredCommandStorage` object.
+
+9. Within the `setStorage` method, various method calls are made for each module category (Eg. Foundation, IT Professionalism)
+ to set up the `RequiredCommandStorage` object with the all the relevant modules.
+
+10. Then, `RequireCommand.execute()` call its own method `compareAllGEs()` to check if any GE pillars have not been cleared.
+It then keeps track of which pillars have not been cleared.
+
+11. `RequireCommand.execute()` then proceeds to call its own methods `compareModules`, `compareScience` and `compareInternship`
+to keep track of undone modules by cross-referring to the `currentModules` list.
+
+12. Finally, a `CommandResult` is created with the `leftOverModules` to show the filtered list of remaining required modules.
+
+The following sequence diagram illustrates how the `required` command is executed.
+
+![RequiredDiagram](images/RequiredSequenceDiagram.png)
+
+### Check all available Science Modules
+
+The `science` command allows users to view all available Science modules available on the Computer Science curriculum.
+
+When the command is executed, a list of all available Science modules will be displayed on the `Command Line Display`.
+
+This is achieved by tapping into the `RequiredCommandStorage` class to extract and parse the Science modules, while the
+`ScienceCommand` class handles the logic of displaying the modules. This command is separated from the `required`
+command to avoid cluttering of the `Command Line Display`. 
+
+As with all operations in GradPad, the `ScienceCommand` class handles the execution of `science` operations.
+
+Given below is a series of steps to show how a `science` operation behaves during its execution.
+
+1. The user enters the `science` command string.
+
+2. This calls the `execute` method of the `LogicManager` class with the user input passed in as a string.
+
+3. `Logic.execute()` then calls the `parseCommand`  method of the `GradPadParser` class to parse the string input.
+
+4. `GradPadParser.parseCommand()` identifies the command as a science command and thus creates a `ScienceCommand`
+object.
+
+5. This command object is then passed back to the `LogicManager` in step 2.
+
+6. `LogicManager` executes the newly created `ScienceCommand`, which will contain a list of `scienceModules` hard coded in.
+
+7. Then, the `ScienceCommand.execute()` calls its own method, `setScienceModules` which sets the `scienceModules` with 
+the list of available modules.
```

Comment:
Actually `ScienceCommand` doesn't contain a list of hard-coded modules right? HAHA if not then it wouldn't have to set and get the science modules from`RequiredCommandStorage`

Reply:
oh right my bad....HAHAH forgot to update accordingly. Updated!